eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Deleg.hs

```diff
@@ -66,7 +66,7 @@ data ConwayDelegPredFailure era
   = IncorrectDepositDELEG !Coin
   | StakeKeyAlreadyRegisteredDELEG !(Credential 'Staking (EraCrypto era))
   | StakeKeyNotRegisteredDELEG !(Credential 'Staking (EraCrypto era))
-  | StakeKeyHasNonZeroAccountBalanceDELEG !Coin
+  | StakeKeyHasNonZeroAccountBalanceRewardDELEG !Coin
```

Comment:
I'm not sure if adding `Reward` at the end makes anything clearer. I'm not sure what the predicate failure means either, so maybe it's indeed necessary.

Reply:
hmm or maybe it'd be clearer if it was `StakeKeyHasNonZeroRewardAccountBalanceDELEG`

Reply:
:raised_hands: